### PR TITLE
llvmdev recipe: Add patch that clears GOTOffsetMap

### PR DESCRIPTION
**Pull request by gmarkall**
See:

- https://github.com/llvm/llvm-project/issues/61402
- https://github.com/numba/numba/issues/8738

Fixes numba/numba#8738.

### CODE REVIEW

File: conda-recipes/llvmdev/meta_llvm14.yaml
Comment (stuartarchibald) [inline conda-recipes/llvmdev/meta_llvm14.yaml:1]: Think this patch should cause an increment to the `build_number`, i.e. 0 -> 1.

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch (and great debugging/fixing) @gmarkall. I think that perhaps the best proof of this as a fix will come from testing against the Numba stack, though the commentary in https://github.com/numba/numba/issues/8738 is by itself certainly compelling. One minor comment to resolve else looks good for testing.

File: conda-recipes/llvmdev/meta_llvm14.yaml
Comment (gmarkall) [reply conda-recipes/llvmdev/meta_llvm14.yaml:1]: Doh - fixed!

Comment (stuartarchibald) [reply conda-recipes/llvmdev/meta_llvm14.yaml:1]: Many thanks!

**Comment by gmarkall** [comment]
@stuartarchibald Version number bumped - I missed changing the status of this PR earlier, apologies.

**Pull request merged**